### DISCUSSION TITLE
How to access full bricks function list on self-hosted refinery

### DISCUSSION THREAD

**Discussion by Analect** [code-kern-ai/refinery#265] [Q&A] [answered]
I have been playing around with a self-hosted Refinery today. Overall a really great experience, so thanks.

However, I was experimenting around leveraging Bricks from within the 'adding attributes' screen. I'm not sure if the self-hosted bricks, documented [here](https://github.com/code-kern-ai/bricks#getting-started) is intended to work in conjunction with the self-hosted refinery? I was looking at the [Named Entity Recognition with Kern AI refinery, bricks & Python](https://www.youtube.com/watch?v=hFOk76vCLGY) video, and tied to find `Person extraction` from within my self-hosted refinery, but it's not visible, per the screenshot below. I'm running a self-hosted `refinery v.1.11.0` and self-hosted `bricks FastAPI 0.1.0 OAS3`.

![image](https://github.com/code-kern-ai/refinery/assets/4063815/8d3479ee-5a50-4123-8c98-e078ac806fec)

If I click on 'Group Filter' (from within the Bricks Integrator dialogue-box) it suggests 44 items, whereas there are more like 88 exposed in the self-hosted bricks (see list lower down) and maybe 90 suggested at [bricks.kern.ai/home](https://bricks.kern.ai/home).

![image](https://github.com/code-kern-ai/refinery/assets/4063815/9a24e594-a20d-4fdc-92bf-0272017ba62c)

What am I missing ... so that I can access a longer list of bricks?  Is there documentation around how a locally-hosted bricks is called / integrated from a self-hosted refinery?

Thanks for any guidance.

```
# 88 functions per Schemas in http://localhost:8000/docs (FastAPI on locally hosted Bricks)
AddressExtractionModel
AspectExtractionModel
AzureSpeechToTextModel
BertNerExtractionModel
BertSentimentGermanModel
BertToxicityDetectorModel
BicExtractionModel
BingNewsSearchModel
BingSearchModel
BingSpellingCorrectionModel
ColorCodeExtractionModel
CosineSimilarityModel
CreditCardExtractionModel
DateExtractionModel
DebertaNerExtractionModel
DebertaReviewClassifierModel
DeeplTranslatorModel
DifficultWordsExtractionModel
DigitExtractionModel
DistilbertStockNewsClassifierModel
DomainParserModel
EmailExtractionModel
EmotionalityDetectionModel
EuclideanDistanceModel
FilepathExtractionModel
GazetteerExtractionModel
GoodbyeExtractionModel
GoogleSearchModel
GptClassifierModel
GptGrammarCorrectionModel
GptInformationExtractionModel
GptTldrSummarizationModel
HTTPValidationError
HammingDistanceModel
HashtagExtractionModel
HtmlCleanserModel
HtmlUnescapeModel
IbanExtractionModel
IbmTranslatorModel
IpExtractionModel
IsbnExtractionModel
KeywordExtractionModel
LanguageDetectionModel
LanguageTranslatorModel
LevenshteinDistanceModel
LookupListModel
ManhattanDistanceModel
MetricExtractionModel
MicrosoftTranslatorModel
MostFrequentWordsModel
NounMatchExtractionModel
NytNewsSearchModel
OrgExtractionModel
PartOfSpeechExtractionModel
PercentageExtractionModel
PersonExtractionModel
PhoneNumberExtractionModel
PhoneticSoundexModel
PriceExtractionModel
ProfanityDetectionModel
QuoteExtractionModel
ReadingTimeModel
RegexExtractionModel
SentenceComplexityModel
SmalltalkExtractionModel
SmalltalkTruncationModel
SpacyLemmatizerModel
SpellingCheckModel
StockTickerExtractionModel
SubstringExtractionModel
SumyWebsiteSummarizerModel
SyllableCountModel
SynonymExtractionModel
TextSummarizationModel
TextblobSentimentModel
TextblobSpellingCorrectionModel
TextblobSubjectivityModel
TimeExtractionModel
UrlExtractionModel
UrlKeywordParserModel
VaderSentimentClassifierModel
VaderSentimentScoresModel
ValidationError
VerbPhraseExtractionModel
WindowSearchExtractionModel
WorkOfArtExtractionModel
WorkdayClassifierModel
ZipcodeExtractionModel
```

**Comment by LeonardPuettmannKern** [comment, marked as answer]
Hi @Analect, thanks for your question! You actually don't need to self-host bricks on your computer, as the self-hosted version of refinery works fine with all the bricks hosted via bricks.kern.ai. The reason why you are not finding the person extraction brick is that this specific brick (an extraction module) is not meant to work with the attribute calculation, but only as a labeling function (which you can create on the heuristics page). In general, all extraction modules only work with labeling functions. Most classifier and generator modules (think translation or text cleaning for example) work great for the attribute calculation. 

You can learn more about the types of bricks in the bricks repository https://github.com/code-kern-ai/bricks or the official docs here: https://docs.kern.ai/bricks/refinery-integration. If the differentiation is not clear enough we are happy to enrich the docs of course. 

Let me know if that helped. If you are still struggling we can also jump into a quick call otherwise! :) Also, we're always open to contributions for new bricks, even if it's just ideas or suggestions for new brick modules. 

- Leo

**Reply by Analect** [reply]
Thanks @LeonardPuettmannKern for the quick response.
